src/data/nat/pairing.lean

```diff
@@ -132,3 +132,35 @@ lemma Union_unpair_prod {α β} {s : ℕ → set α} {t : ℕ → set β} :
 by { rw [← Union_prod], convert surjective_unpair.Union_comp _, refl }
 
 end set
+
+section complete_lattice
+
+lemma supr_unpair_sup {α} [complete_lattice α] {s t : ℕ → α} :
+  (⨆ n : ℕ, s n.unpair.1 ⊔ t n.unpair.2) = (⨆ i : ℕ, s i) ⊔ ⨆ j, t j :=
+begin
```

Comment:
Maybe the better statement here is
```suggestion
lemma supr_unpair {α} [complete_lattice α] {f : ℕ → ℕ → α} :
  (⨆ n : ℕ, f n.unpair.1 n.unpair.2) = (⨆ i j, f i j) :=
begin
```
which avoids the need to have four lemmas

Reply:
can we get `supr_unpair_sup` from `supr_unpair`? I don't think lean can figure out what  `f` is

Reply:
It works if we make `f` explicit so I think we should do that